woocommerce/woocommerce-android
# origin/issue/5962-product-detail-bottom-sheet-exception-take2

This is the second attempt to fix #5962, a mystery crash in `ProductDetailBottomSheetAdapter`. I noticed that rather than return a unique ID in that adapter's `getItemId()`, we return `position.toLong()`. We do that in many places in WCAndroid, but ideally if an adapter has stable IDs we should provide a unique ID instead of the position.

So, this PR updates the adapter to use a unique ID for each item. I also slightly modified the way the list is populated, but that was for clarity rather than an attempt to fix this problem.

If we continue to see these crashes after this PR is in a release, I recommend the next step is to try not using unique IDs for the adapter.

To test:
 * Open product detail
 * Tap the "Add more details button"
 * Ensure the bottom sheet works as expected
